### PR TITLE
Update GeckoView Nightly to 66.0.20190111093148.

### PR DESCRIPTION
**Pull request by pocmo**
~~Blocked on a Java/Kotlin API compatibility problem.~~

### CODE REVIEW

**Comment by pocmo** [comment]
Still need to update some tests.

**Comment by pocmo** [comment]
Now the test should pass.

@csadilek This touches the settings code. Can you give that a review?

I think previously we were creating a GeckoSession twice: once when initializing the property `geckoSession` and a second time in `initGeckoSession`. This should be fixed now.

We lost the `settingPrivateMode` test unfortunately. Private mode can't be set on the settings anymore (and was not expected to ever be supported there) but needs to be set from a settings builder before creating the session. Maybe we can refactor this in a way that we can rewrite such a test.

File: components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt [hunk 1]
```diff
@@ -1107,7 +1098,7 @@ class GeckoEngineSessionTest {
         GeckoEngineSession(runtime, geckoSessionProvider = geckoSessionProvider,
                 privateMode = false, defaultSettings = defaultSettings)
 
-        verify(geckoSession.settings).setBoolean(GeckoSessionSettings.USE_TRACKING_PROTECTION, true)
+        assertFalse(geckoSession.settings.usePrivateMode)
```
[the remark this answers is not in the document]
Comment (csadilek) [reply components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt]: And, of course, there was one missing before :)

[the remark this answers is not in the document]
Comment (pocmo) [reply components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt]: Okay, I'll add that :)

**Pull request merged**